### ISSUE TITLE
NextJS: "Warning: useLayoutEffect does nothing on the server"

### ISSUE THREAD

**Issue by elte0** [iiroj/use-breakpoint#10]
Hey there!

When using useBreakpoint with NextJS (SSR), I'm getting this warning:
`Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.`

I wrote a helper for setting the bootstrap breakpoints:

```
import useBreakpoint from "use-breakpoint"

const BREAKPOINTS = {
  xs: 0,
  sm: 576,
  md: 768,
  lg: 996,
  xl: 1200,
  xxl: 1400
}

function useBootstrap() {
  return useBreakpoint(BREAKPOINTS, "md")
}

export default useBootstrap

```

and then use it in my component like this:
```
const { breakpoint } = useBootstrap()
```

Am I missing something?

**Comment by iiroj** [comment]
Hello,

this hook does indeed use the `useLayoutEffect` hook internally, and since hooks cannot be called conditionally, the error is currently unavoidable.

It is safe to ignore since this hook doesn't really do anything server-side except return the optional  default value.

See this PR for more info: https://github.com/iiroj/use-breakpoint/pull/9

**Comment by elte0** [comment]
I see, thanks for the answer :)

**Comment by amannn** [comment]
Would it be an option to make use of [use-isomorphic-layout-effect](https://www.npmjs.com/package/use-isomorphic-layout-effect)? With static site generation this currently floods my console quite heavily.

**Comment by iiroj** [comment]
@amannn I don't think a dependency is necessary for this, just a simple `if (typeof window === "undefined")`. I can update it when I have the time, or a PR is also welcome!

**Comment by iiroj** [comment]
I just released [2.0.2](https://github.com/iiroj/use-breakpoint/blob/master/CHANGELOG.md#202-2021-08-30) that will hopefully suppress the error. Let me know if it doesn't work!

**Comment by amannn** [comment]
Works perfectly, thanks!
